### ISSUE TITLE
Adding comparison to UUID

### ISSUE THREAD

**Issue by jwoertink** [crystal-lang/crystal#11023]
Should UUIDs be comparable so `Array(UUID)` could be sorted alphanumerically? 

```crystal
require "uuid"

uuids = [UUID.new("84d9906d-b124-47b8-9632-a54ad7b43c06"), UUID.new("01cd1369-dd6e-4b3c-8131-2836dfe6102c")]

sorted = uuids.sort
```

```
 199 | v = v1 <=> v2
              ^--
Error: undefined method '<=>' for UUID
```

I'm assuming this wasn't originally added because UUIDs aren't normally sorted, but if you only need them in alphanumerical order, then right now you have to map them to strings. However, if you still need them as UUID, then you would have to re-map them back to UUID after sorting.

```crystal
uuids.map(&.to_s).sort.map {|id| UUID.new(id)}
```

**Comment by asterite** [comment]
sort_by &.to_s

?

**Comment by jwoertink** [comment]
I guess that works too. Would there be a way to point people to using that in this case?

```
uuids.sort

Error: To sort Array(UUID) use `sort_by &.to_s` instead
```

It may just be me, but using `sort_by` didn't even cross my mind 😅

**Comment by Sija** [comment]
@jwoertink This looks more like a pretty typical RTFM situation ;)

**Comment by asterite** [comment]
That said, maybe uuid could be sortable. We could check what other languages do.

**Comment by Blacksmoke16** [comment]
Looks like Java does it: https://github.com/openjdk/jdk16/blob/master/src/java.base/share/classes/java/util/UUID.java#L496-L521.

Related https://stackoverflow.com/questions/51031195/why-java-util-uuid-is-comparable.

**Comment by vlazar** [comment]
@jwoertink What's your use case to sort UUID?

Maybe sortable UUID should be a thing on its own?

There are these for example:
https://github.com/SuperPaintman/ulid
https://github.com/Sija/ksuid.cr
https://github.com/wyhaines/csuuid.cr

**Comment by jwoertink** [comment]
I'm creating a composite key based on the standard UUID where they are in alphabetical order. This is so I can join 2 records in the DB where their primary keys are UUID, and I don't end up with duplicate relations where they are in different orders. If they're always in alphabetical order, then I can guarantee no duplicates.

**Comment by straight-shoota** [comment]
Resolved by #11352
